### DISCUSSION TITLE
Getting a lot of 'Could not unserialize response, returning uncached response instead.'

### DISCUSSION THREAD

**Discussion by chrispage1** [spatie/laravel-responsecache#439] [General]
Hi,

I'm getting a lot of errors throwing up 'Could not unserialize response, returning uncached response instead.'. While I don't think this is necessarily an issue, I'd be keen to know the reason why I might be getting this?

Thanks,
Chris.

**Comment by jarnovanleeuwen** [comment]
This happens when the response has been cached, but is cleared or expired between calling `hasBeenCached` and `getCachedResponseFor`. Thanks to https://github.com/spatie/laravel-responsecache/pull/434, this situation is handled correctly and simply treated as a `CacheMiss`. However, it still reports when this happens.

You can add the following code to Laravel's exception handler to suppress those errors.

```php
$this->reportable(function (Exception $exception) {
    /**
     * laravel-responsecache correctly handles a race condition when the cached response is cleared or expired
     * between calling 'hasBeenCached' and 'getCachedResponseFor'. However, it still does report when this happens.
     * Because everything continues to work in this situation, and there is no bug, we are ignoring this message.
     *
     * See https://github.com/spatie/laravel-responsecache/pull/434.
     */
    if ($exception->getMessage() === 'Could not unserialize response, returning uncached response instead. Error: Could not unserialize serialized response ``') {
        return false;
    }
});
```
